Fix scrolling image view when using arrow keys

Stubenhocker1399 opened the pull request:
Hi! :wave: 

This fixes two bugs when switching between multiple image views using arrow keys; one being sometimes the modal not showing a scroll bar (and not being scroll-able) even though it is too tall to fit, and the other not being able to scroll using arrows up and down.

As a note; changing that the next modal only opens on `onClosed` does not seem like the best fix for the scroll bar bug, as it seems to be an underlying bug in the general modal code and potentially is already an issue in other places as well.

Stubenhocker1399 committed:
Fix scrolling image view when using arrow keys

dominikklein commented:
@Stubenhocker1399 Thanks for the input. When this is related to a bug, can you please create a related issue with some information to reproduce the problem?

Stubenhocker1399 commented:
Okay, sure! :smiley:

fliebe92 commented:
Hey @Stubenhocker1399, thanks for your pull request 😄 

We checked it and right now we're facing two issues:

1. You are using `this.nextElement`, but we encourage to simply use `@nextElement` in our CoffeeScript files.
2. It seems this pull request is introducing a new bug, by opening the image preview for one of your provided test images (special thanks for this), the dialog is automatically closed once. This only happens after a page reload, every attempts to open the image preview are fine afterwards.

We would really appreciate if you could check and improve your pull request.

Hope to read you soon! 😎

Stubenhocker1399 committed:
Adjust code style

Stubenhocker1399 commented:
Glad you found the test images useful :slightly_smiling_face: 

Regarding 2.; are you sure that is a new bug? At least, if I wait for the page to fully load (including any sidebar tickets), I have no automatic modal closing triggered even with the new changes. It is only an issue when there are tickets on the side bar still loading, which once they load trigger all modals to be closed. This however is already the same behaviour without my changes.

(It is caused by this line hiding all modals https://github.com/zammad/zammad/blob/8ea22f28ebb439e6b4b5ac330032199497153594/app/assets/javascripts/app/controllers/ticket_zoom.coffee#L340; triggered by the finished loading ticket zoom controller in the background being hidden and calling the above https://github.com/zammad/zammad/blob/8ea22f28ebb439e6b4b5ac330032199497153594/app/assets/javascripts/app/lib/app_post/task_manager.coffee#L325-L327)

If it is not this particular bug, then please let me know how to reproduce it, as I don't have any automatic modal closing happening for me otherwise.

Of course the above is an issue by itself and should preferably be fixed; even if it is already an existing bug. :smile:

Stubenhocker1399 committed:
Fix loading sidebar tickets always closing modals

Only close when the ticket zoom view was actually shown at least once,
as otherwise it might be one that does not stem from this controller.

Stubenhocker1399 commented:
I've pushed a change that seems to fix that particular issue while still closing any modals correctly as described in #2104.

If there are any further issues let me know. :smiley:

fliebe92 commented:
Good morning @Stubenhocker1399 😎 

Thanks for your time to check my last comment and do some further fixing. I was not able to reproduce the described behavior without your changes, but maybe it was a timing issue or something else. I am going to check your changes soon to get this merged. Stay tuned!

Thanks for your patience and for improving Zammad! ❤️

fliebe92 commented:
Hey @Stubenhocker1399, your changes are fine and merged to `develop` by now, `stable` will follow soon!

Thanks again for your pull request, we really appreciate it! ❤️
